java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitType.java

```diff
@@ -152,16 +155,25 @@ private static boolean hasDiagnosticErrors(CompilationInfo info, Tree tree) {
     private static boolean isValidType(HintContext ctx) {
         TreePath treePath = ctx.getPath();
         TreePath initTreePath = ctx.getVariables().get("$init");  //NOI18N
+        TypeMirror variableTypeMirror = ctx.getInfo().getTrees().getElement(treePath).asType();
 
         if (initTreePath.getLeaf().getKind() == Tree.Kind.NEW_CLASS) {
             NewClassTree nct = ((NewClassTree) initTreePath.getLeaf());
             if (nct.getClassBody() != null) {
                 return false;
             }
+        } else if (initTreePath.getLeaf().getKind() == Tree.Kind.METHOD_INVOCATION) {
+            //filter anonymous class type return type
+            if (variableTypeMirror.getKind() == TypeKind.DECLARED) {
+                DeclaredType dt = (DeclaredType) variableTypeMirror;
+                TypeElement typeElem = (TypeElement) dt.asElement();
+                ClassSymbol sym = (ClassSymbol) typeElem;
+                if (sym.isAnonymous()) {
```

Comment:
Please do not use internal javac classes outside of the infrastructure (ideally, outside of java.source* and lib.nbjavac, although I know it is used in spi.java.hints as well).

In this case typeElem.getNestingKind() == NestingKind.ANONYMOUS should work, right?

Reply:
Corrected.